---
author: Aditya Singh
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
    md_document:
    variant: gfm
        bibliography: bibliography.ris
        ---
          
## Overview
              
The two main steps in performing differential expression analysis are to
estimate the relative abundance of transcripts, and to apply statistical models
to test for differential expression between treatment groups. Estimating
relative abundance is basically determining how many NGS reads match a given
gene within a genome. In this module you'll use Salmon [@Patro] to estimate
relative abundance, tximport [@Soneson] to import the Salmon abundance
estimates, and DESeq2 [@Love] to perform statistical tests to identify
differentially expressed genes. 

## References
Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.” *Genome Biol* 15 (12): 550–50..

Patro, Rob, Geet Duggal, Michael I. Love, Rafael A. Irizarry, and Carl Kingsford. 2017. “Salmon Provides Fast and Bias-Aware Quantification of Transcript Expression.” *Nat Methods* 14 (4): 417–19..

Soneson, Charlotte, Michael I. Love, and Mark D. Robinson. 2016. “Differential Analyses for RNA-Seq: Transcript-Level Estimates Improve Gene-Level Inferences.” *F1000Res* 4 (February): 1521–1.

